## /bin/sh -e simulate dpatch for prepend_chlog.py
## correct_ps_display.dpatch by Kilian Krause <kilian@debian.org>
## DP: Display the correct pid in the asterisk console. Closes: 338646
## DP: Upstream bug: #7098 . Fixed in 1.4 and in 1.2 branch (remove on 1.2.8)

Index: asterisk-1.2.27/asterisk.c
===================================================================
--- asterisk-1.2.27.orig/asterisk.c	2008-03-19 11:33:05.000000000 +0100
+++ asterisk-1.2.27/asterisk.c	2008-03-19 11:33:24.000000000 +0100
@@ -2324,11 +2324,12 @@
 			(!option_verbose && !option_debug && !option_nofork && !option_console)
 	) {
 		daemon(0,0);
+		ast_mainpid = getpid();
 		/* Blindly re-write pid file since we are forking */
 		unlink((char *)ast_config_AST_PID);
 		f = fopen((char *)ast_config_AST_PID, "w");
 		if (f) {
-			fprintf(f, "%d\n", (int)getpid());
+			fprintf(f, "%d\n", ast_mainpid);
 			fclose(f);
 		} else
 			ast_log(LOG_WARNING, "Unable to open pid file '%s': %s\n", (char *)ast_config_AST_PID, strerror(errno));
